src/vidpf.rs

```diff
@@ -799,6 +776,28 @@ impl<F: FieldElement> ParameterizedDecode<<Self as IdpfValue>::ValueParameter> f
     }
 }
 
+#[derive(Copy, Clone)]
+struct VidpfIndex<'a> {
+    bit: Choice,
+    input: &'a VidpfInput,
+    level: u16,
+}
+
+impl VidpfInput {
+    fn index_iter<'a>(
+        &'a self,
+    ) -> Result<Box<dyn Iterator<Item = VidpfIndex<'a>> + 'a>, VidpfError> {
```

Comment:
I used `Box<dyn ...>` here simply because I didn't wanna try to figure out the type of the iterator :D